addons/account/models/account_reconcile_model.py

```diff
@@ -290,7 +290,7 @@ def _get_taxes_move_lines_dict(self, tax, base_line_dict):
                 tax_result['tag_ids'] = self.env['account.move.line']._revert_signed_tags(tax_tags).ids
 
         new_aml_dicts = []
-        for tax_res in res['taxes']:
+        for tax_res in (tax for tax in res['taxes'] if tax['amount']):
```

Comment:
never check if a monetary amount is zero without using `currency.is_zero()` ;)

Reply:
Good to know, thanks for the tip :slightly_smiling_face: